types/govuk-frontend/dist/govuk/govuk-frontend-component.d.ts

```diff
@@ -2,8 +2,38 @@
  * Base Component class
  *
  * Centralises the behaviours shared by our components
- *
- * @internal
  */
-export class GOVUKFrontendComponent {
+export abstract class GOVUKFrontendComponent<RootElementType extends Element = HTMLElement> {
+    static elementType: typeof Element;
+
+    /**
+     * Validates whether components are supported
+     *
+     * @throws {SupportError} when the components are not supported
+     */
+    static checkSupport(): void;
+
+    /**
+     * Constructs a new component, validating that GOV.UK Frontend is supported
+     *
+     * @internal
+     * @param {Element | null} [$root] - HTML element to use for component
+     */
+    constructor($root?: Element | null);
+
+    /**
+     * Returns the root element of the component
+     *
+     * @returns {RootElementType} - the root element of component
+     */
+    protected get $root(): RootElementType;
+}
+
+export interface ChildClass {
+    /**
+     * - The module name that'll be looked for in the DOM when initialising the component
+     */
+    moduleName: string;
 }
+
+export type ChildClassConstructor = typeof GOVUKFrontendComponent & ChildClass;
```

Comment:
@romaricpascal Were these supposed to be public?

Reply:
Ah, nope, they're internal definitions used for some casting in the `GOVUKFrontendComponent` code.

Reply:
No worries, I'll manually drop them for now until the JSDoc is updated

They weren't referenced by anything public (unlike `Config` or `ConfigKey` etc) so wasn't sure

Reply:
✅ Removed ([see diff](https://github.com/DefinitelyTyped/DefinitelyTyped/compare/b00c25c5d4f455db65f68c104199e9401c4fe66c..5e2bfc0e563b0a676adfbe1e946e060a0927bcc5))